### ISSUE TITLE
[ui5-task-zipper] Error For Third Party Dependencies

### ISSUE THREAD

**Issue by Magoli1** [ui5-community/ui5-ecosystem-showcase#540]
Hey everyone,

in our project we are making use of the builder settings of the ui5-tooling to include external dependencies, in this case swagger-ui resources. A snippet of the `ui5.yaml` looks like this:
````
specVersion: "2.5"
metadata:
  name: test-app
type: application
builder:
  settings:
    includeDependency:
      - swagger-ui-dist
  resources:
    excludes:
      - /test/**
      - /localService/**
      - /test.html
  customTasks:
    - name: ui5-task-zipper
      afterTask: generateCachebusterInfo
      configuration:
        archiveName: ui
        includeDependencies: true
---
specVersion: "2.5"
kind: extension
type: project-shim
metadata:
  name: swagger-lib
shims:
  configurations:
    "swagger-ui-dist":
      specVersion: "2.5"
      type: module
      metadata:
        name: swagger-ui-dist
      resources:
        configuration:
          paths:
            /libs/swagger/: ""
````

During the zip-task, the following error comes up:

``
builder:customtask:zipper Couldn't add all resources to the archive: Error: absolute path: /libs/swagger/package.json
``

This is probably due to this string replacement, as here only `/resources` and `/test-resources` get replaced, but not all possible strings (eg. in this case (/libs/swagger/**):
https://github.com/ui5-community/ui5-ecosystem-showcase/blob/899e3998ee47efd545aa85205e3cd01dbfbd6cc9/packages/ui5-task-zipper/lib/zipper.js#L57

I will provide a possible fix for this. Let me know your thoughts 😁 

BR Robert

**Comment by LukasHeimann** [comment]
I am experiencing the same problem. Additionally, though, I find that the simple flag `includeDependencies` is not enough for me, as I don't do a standalone build (so I don't want to include the ui5 dependencies), but I still need my `project-shim`s.

I'm using the feature described here to achieve this: https://github.com/SAP/ui5-tooling/issues/494

```yaml
builder:
  settings:
    includeDependency:
      - ext1
      - ext2
  customTasks:
    - name: ui5-task-zipper
      afterTask: generateCachebusterInfo
      configuration:
        archiveName: gacd
        includeDependencies: true # should be using the general includeDependency setting
        additionalFiles:
          - xs-app.json
```

**Comment by mauriciolauffer** [comment]
Because you're zipping the resources, I assume you're going to deploy the project. The [ui5-deployer](https://www.npmjs.com/package/ui5-deployer) could help you with it either doing the deploy via ADT or OData, no need for ui5-task-zipper.

**Comment by LukasHeimann** [comment]
I need to zip to publish to the HTML5 repository of SAP's BTP via an MTA deployment... So I actually do need to build a ZIP archive.

**Comment by Magoli1** [comment]
As this has been implemented in #541 , I will recheck and then close this issue 😄

**Comment by LukasHeimann** [comment]
I retested my project and it looks good! Thank you for implementing this!
